packages/block-editor/src/store/selectors.js

```diff
@@ -1481,13 +1491,74 @@ export const getBlockInsertionPoint = createSelector(
 		return { rootClientId, index };
 	},
 	( state ) => [
-		state.insertionPoint,
+		state.insertionCue,
 		state.selection.selectionEnd.clientId,
 		state.blocks.parents,
 		state.blocks.order,
 	]
 );
 
+/**
+ * Returns the insertion cue state. Returns null if there is no insertion cue.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {Object} Insertion point object with `rootClientId`, `index`.
+ */
+export function getInsertionCue( state ) {
+	return state.insertionCue;
+}
+
+/**
+ * Returns the point where a block would be inserted.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {Object} Insertion point object with `rootClientId`, `index`.
+ */
+export const getNextInsertionPoint = createRegistrySelector( ( select ) =>
+	createSelector(
+		( state ) => {
+			let rootClientId, index;
+
+			const {
+				insertionPoint,
+				selection: { selectionEnd },
+			} = state;
+			if ( insertionPoint !== null ) {
+				return insertionPoint;
+			}
+
+			const { clientId } = selectionEnd;
+			const sectionRootClientId = unlock(
+				select( STORE_NAME )
+			).getSectionRootClientId( state, clientId );
+
+			if ( clientId ) {
+				rootClientId =
+					getBlockRootClientId( state, clientId ) || undefined;
+				index = getBlockIndex( state, selectionEnd.clientId ) + 1;
+			} else if ( sectionRootClientId ) {
+				index = getBlockOrder( state, sectionRootClientId ).length;
+			} else {
+				index = getBlockOrder( state ).length;
+			}
+
+			return { rootClientId, index };
+		},
+		( state ) => [
+			state.insertionPoint,
+			state.selection.selectionEnd.clientId,
+			state.blocks.parents,
+			state.blocks.order,
+		]
+	)
+);
```

Comment:
This is only used in one spot, so I think we should remove it. I think it could be useful in the future though.

Reply:
I also think that if we had it available, it would have been used. May be worth going through and seeing how much it would simplify things.